### PR TITLE
Make time shift corrections optional

### PR DESCRIPTION
**Pull request by maxnoe**
This adds boolean telescope options to the `CameraCalibrator` for applying the time shifts.

Waveform shifting and peak time shifting can be independently turned on / off.

### CODE REVIEW

File: ctapipe/calib/camera/tests/test_calibrator.py [hunk 1]
```diff
@@ -177,6 +177,21 @@ def test_dl1_charge_calib(example_subarray):
         event.dl1.tel[telid].peak_time, mid / sampling_rate, atol=1
     )
 
+    # test deactivating timing corrections
+    calibrator.apply_waveform_time_shift = False
+    calibrator(event)
+    # times should still be correct, since peaks are shifted
+    np.testing.assert_allclose(
+        event.dl1.tel[telid].peak_time, mid / sampling_rate, atol=1
```
Comment (watsonjj) [inline ctapipe/calib/camera/tests/test_calibrator.py:185]: Same result as with timing corrections activated - so not really a useful test

File: ctapipe/calib/camera/tests/test_calibrator.py [hunk 2]
```diff
@@ -177,6 +177,21 @@ def test_dl1_charge_calib(example_subarray):
...
+    )
+
+    calibrator.apply_peak_time_shift = False
+    calibrator(event)
+    # no we should be back to the result without time shift
```
Comment (watsonjj) [inline ctapipe/calib/camera/tests/test_calibrator.py:190]: spelling "now"

File: ctapipe/calib/camera/tests/test_calibrator.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/calib/camera/tests/test_calibrator.py:185]: Yes, but I don't have a good idea to test this.

Do you have a proposal? Maybe using the global peak integrator with large timeshifts?

Comment (maxnoe) [reply ctapipe/calib/camera/tests/test_calibrator.py]: I now added tests using the `GlobalPeakWindowSum` that demonstrate nicely that the option works and why the peak time shifting is important in general.

File: ctapipe/calib/camera/tests/test_calibrator.py (hunk 2 above)
Comment (maxnoe) [reply ctapipe/calib/camera/tests/test_calibrator.py]: fixed

**Review by kosack — APPROVED** [review]

**Comment by maxnoe** [comment]
@watsonjj I implemented a better test, can you re-review?

**Review by watsonjj — DISMISSED** [review]

**Comment by maxnoe** [comment]
Merge conflicts resolved, plase review again @kosack @watsonjj

**Review by watsonjj — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**